R/compare.R

```diff
@@ -314,6 +329,55 @@ expand_diff_text <- function(old, new) {
   list(old = old, new = new)
 }
 
+merge_packages <- function(old, new) {
+  opkgs <- parse_pkgs(old)
+  npkgs <- parse_pkgs(new)
+
+  if (is.null(opkgs) || is.null(opkgs)) return(list(old = old, new = new))
+
+  names_to_keep <- c("package", "version", "source")
+  opkgs$pkgs <- opkgs$pkgs[names_to_keep]
+  npkgs$pkgs <- npkgs$pkgs[names_to_keep]
+  names(opkgs$pkgs) <- c("package", "old_version", "old_source")
+  names(npkgs$pkgs) <- c("package", "new_version", "new_source")
```

Comment:
How worried do I need to be that these 3 columns are not present in both sessioninfo inputs?

Reply:
It may happen theoretically, if you explicitly request what to show and leave out the package versions. But it should be rare I guess, so I think we can just error out of it does. Or fall back to the regular diff?